### DISCUSSION TITLE
Change Language on Webpage?

### DISCUSSION THREAD

**Discussion by silicahd** [givanz/Vvveb#68] [General]
Maybe I am doing it wrong. I am trying to create a site in English and French. Lets say you make the home page on English how do you work on the French one? Or same with the other pages? In admin under pages it is only showing the English.

**Comment by givanz** [comment]
To translate texts from html templates such as texts for homepage hero you need to add translations for the text in the language pack, gettext po files.

In the latest.zip update there is a [language translations editor](http://demo.vvveb.com/admin/?module=localization/languages) to change or add new translations, changed files /system, /admin and /public/admin.

By default there are 3 files/domains, `vvveb` for admin, `landing-theme` for frontend/default theme  and `user` that is used to add custom translations or to keep translations when the language pack is updated. 

When saving `vvveb` or `landing-theme` domains the translations from `user` will be appended/override existing translations before compiling the file.

If you need to have completely different pages for other languages like different layouts then you can use the `language-specific-template` bundled plugin, you can also download it from [language-specific-template.zip](https://github.com/givanz/Vvveb/files/13895524/language-specific-template.zip)

The plugin will check for language specific html file like `index.fr_FR.html` or `content/post.fr_FR.html` before serving general `index.html` and `content/post.html`

You can use save as template or clone from templates list to duplicate pages for other languages and add the language code suffix.

![clone](https://github.com/givanz/Vvveb/assets/302738/fb39d6e3-c156-43b7-8d8d-b3ec387a0ed2)


[Save as template](https://private-user-images.githubusercontent.com/302738/291922862-6838d261-6f6f-4edc-96ce-fff20e3effc9.webm?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5Mjk5NDEsIm5iZiI6MTcwNDkyOTY0MSwicGF0aCI6Ii8zMDI3MzgvMjkxOTIyODYyLTY4MzhkMjYxLTZmNmYtNGVkYy05NmNlLWZmZjIwZTNlZmZjOS53ZWJtP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTBUMjMzNDAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjYwZTE5ZjVjODkyNTc4ZGRiMzFmYzZiMTBlNGMwYzBmOTIyOTQ4ZjdiZmUwNDk5NjE4MTQ4N2FmNTBkMTZmOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.NI5Vcmc66MgSgU0eYXnc1KA-GMKL7xZDZVWMM6avSjQ)

**Comment by silicahd** [comment]
Thank you for posting this. Would it be better if this was the case. Where you can select the language you want and edit the each language? 

![image](https://github.com/givanz/Vvveb/assets/7761519/49509ac7-504d-4f28-9bc0-35a51f663854)

**Comment by givanz** [comment]
This is already implemented for pages, posts and products, only translations were missing that was added in the last update.

After enabling additional languages from settings > localization > languages the new language tabs will show on page, post and product edit page.
I enabled more languages on the demo https://demo.vvveb.com/admin/?module=content/post&post_id=7&type=page

Currently you need to clear cache after enabling languages, I will add automatic clear cache after adding languages in the next update along with copy/clone description from main language.
